### PR TITLE
HIP-1010: Update Token Custom Fee Schedules via Smart Contracts

### PR DESCRIPTION
**Pull request by stoyanov-st**
**Description**:
This PR describes the updates needed to support the custom fee updates via smart contracts.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by lukelee-sl — DISMISSED** [review]
LGTM.

File: HIP/hip-xxxx.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+| 0xe780c5d3 | function updateFungibleTokenCustomFees(address token,  IHederaTokenService.FixedFee[] memory fixedFees, IHederaTokenService.FractionalFee[] memory fractionalFees) returns (int64 responseCode) | ResponseCode | The response code from the call |
+| 0x01f9eb7d | function updateNonFungibleTokenCustomFees(address token, IHederaTokenService.FixedFee[] memory fixedFees, IHederaTokenService.RoyaltyFee[] memory royaltyFees) returns (int64 responseCode) | ResponseCode | The response code from the call |
+
+## Backwards Compatibility
+
+Backwards compatibility is ensured as no existing features are modified. Similar to HTS system contract this HIP simply exposes HAPI entity functionality.
+
+## Security Implications
+
+The newly added flows will adopt the HAPI authorization logic. In this way the transaction would be successful only if the given contract making the call is set as a `feeScheduleKey` to the token.
```
Comment (lukelee-sl) [inline HIP/hip-xxxx.md:69]: I think add a note in here that says that the `feeScheduleKey` must either be the contract id of the calling contract or a 1 of N threshold key would help clarify what will be needed in order to to complete this call.  Please check that this is what the dispatch to the token service will require.

**Review by Nana-EC — APPROVED** [review]
LG, left some header edit suggestions

File: HIP/hip-xxxx.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+last-call-date-time:
+created: 2024-07-11
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/pull/1010
+updated: 2024-07-15
+requires: 18, 206, 514
+requested-by:
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply HIP/hip-xxxx.md]: Edited

File: HIP/hip-xxxx.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+title: Update Token Custom Fee Schedules via Smart Contracts
+author: Stanimir Stoyanov <stanimir.stoyanov@limechain.tech>
+working-group: Nana Essilfie-Conduah <nana@swirldslabs.com>, Luke Lee  <luke.lee@swirldslabs.com>,  David Bakin <david.bakin@swirldslabs.com>, Mustafa Uzun <mustafa.uzun@limechain.tech>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Draft
+last-call-date-time:
+created: 2024-07-11
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/pull/1010
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply HIP/hip-xxxx.md]: Edited

**Review by Nana-EC — DISMISSED** [review]

**Review by Nana-EC — DISMISSED** [review]

**Review by Nana-EC — APPROVED** [review]

**Comment by AlbertoMolinaIoBuilders** [comment]
LGTM

**Comment by RaphaelMessian** [comment]
LGTM

**Review by RaphaelMessian — DISMISSED** [review]
LGTM

**Review by mgarbs — APPROVED** [review]

**Pull request merged**